### PR TITLE
hardening/1277_add_v1_namemappings_API 

### PR DESCRIPTION
**Pull request by pcoello25**
* Fix issue #1277 
* 100% unit tests passed:
```
Results :
Tests run: 74, Failures: 0, Errors: 0, Skipped: 0  [CYGNUS-COMMON]
Tests run: 249, Failures: 0, Errors: 0, Skipped: 0 [CYGNUS-NGSI]
```
* e2e (unofficial) tests passed:
```
$ curl -X GET http://localhost:8081/v1/namemappings
{"success":"true",
   "serviceMapping": [
      {
         "originalService":"",
         "newService":"",
         "servicePathMappings": [
            {
               "originalServicePath":"",
               "newServicePath":"",
               "entityMappings": [
                  {
                     "originalEntityId":"",
                     "originalEntityType":"",
                     "newEntityId":"",
                     "newEntityType":"",
                     "attributeMappings":[
                        {
                           "originalAttributeName":"",
                           "originalAttributeType":"",
                           "newAttributeName":"",
                           "newAttributeType":""
                        }
                    ]
                  }
                ]
            }
         ]
      }
   ]
}

```
* Assignee: @frbattid

### CODE REVIEW

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/containers/NameMappings.java [hunk 1]
```diff
@@ -54,6 +54,30 @@ public void compilePatterns() {
         } // for
     } // compilePatterns
     
+
```
Comment (frbattid) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/containers/NameMappings.java]: Unnecessary white line (yes, you can hate me :D)

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java [hunk 2]
```diff
@@ -2904,6 +2934,82 @@ private String getGroupingRulesConfFile() throws IOException {
...
+        return nameMappingsConfFile;
+    } // getGroupingRulesConfFile
+    
+    private void loadNameMappings() {
+        String nameMappingsStr = null;
+        
+        // Read the Json string from the configuration file
+        try {
+            nameMappingsStr = JsonUtils.readJsonFile(nameMappingsConfFile);
+            LOGGER.debug("[nmi] Reading name mappings, Json read: " + nameMappingsStr);
```
Comment (frbattid) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java]: These "[nmi]" strings were added in `NGSINameMappingsInterceptor` in roder to say "these are logs written by `NGSINameMappingsInterceptor`". In this class, they are not necessary.

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java [hunk 3]
```diff
@@ -2904,6 +2934,82 @@ private String getGroupingRulesConfFile() throws IOException {
...
+            nameMappings = null;
+            return;
+        } // if
+
+        // Parse the Json string
+        Gson gson = new Gson();
+
+        try {
+            nameMappings = gson.fromJson(jsonStr, NameMappings.class);
+            LOGGER.debug("[nmi] Reading attribute mappings, Json parsed");
```
Comment (frbattid) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java]: Same comment than above...

File: doc/apiary/apiary.apib [hunk 4]
```diff
@@ -119,6 +119,50 @@ Statistics can be reset by a PUT request.
...
+                                {
+                                    "originalEntityId":"",
+                                    "originalEntityType":"",
+                                    "newEntityId":"",
+                                    "newEntityType":"",
+                                    "attributeMappings":[
+                                        {
+                                            "originalAttributeName":"",
+                                            "originalAttributeType":"",
+                                            "newAttributeName":"null",
```
Comment (frbattid) [inline doc/apiary/apiary.apib]: In the .apib examples avoid using "null" or empty values (i.e. try writting something typical such as "Room1" of type "Room" :))

File: doc/cygnus-common/installation_and_administration_guide/management_interface_v1.md [hunk 5]
```diff
@@ -148,7 +149,53 @@ Response:
...
+                            "originalEntityId":"",
+                            "originalEntityType":"",
+                            "newEntityId":"",
+                            "newEntityType":"",
+                            "attributeMappings":[
+                                {
+                                    "originalAttributeName":"",
+                                    "originalAttributeType":"",
+                                    "newAttributeName":"",
+                                    "newAttributeType":""
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface_v1.md]: Same comment than above...

Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface_v1.md]: Fixed in e376b26

File: doc/apiary/apiary.apib (hunk 4 above)
Comment (pcoello25) [inline doc/apiary/apiary.apib]: Fixed in e376b26

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java (hunk 3 above)
Comment (pcoello25) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java]: Fixed in e376b26

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/containers/NameMappings.java (hunk 1 above)
Comment (pcoello25) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/containers/NameMappings.java]: Fixed in e376b26

**Pull request merged**